apps/framework-docs/src/pages/moose/olap/db-pull.mdx

```diff
@@ -0,0 +1,143 @@
+
+<TypeScript>
+```ts filename="app/externalModels.ts"
+// AUTO-GENERATED FILE. DO NOT EDIT.
+// This file will be replaced when you run `moose db pull`.
+
+// ...typed table definitions matching remote EXTERNALLY_MANAGED tables...
+```
+</TypeScript>
+
+### Python output example
+
+<Python>
+```py filename="app/external_models.py"
+# AUTO-GENERATED FILE. DO NOT EDIT.
+# This file will be replaced when you run `moose db pull`.
+
+# ...pydantic models matching remote EXTERNALLY_MANAGED tables...
+```
+</Python>
+
+## Command
+
+```bash filename="Terminal" copy
+moose db pull --connection-string <STRING> [--file-path <OUTPUT_PATH>]
+```
+
+- **--connection-string**: Required. ClickHouse URL (native or HTTP/S)
+- **--file-path**: Optional. Override the default output file. The file at this path will be regenerated (overwritten) on each run.
+
+## Typical Use Cases
+
+### Remote schema changed; update local types
+  Your DBA, CDC pipeline (e.g., ClickPipes), or ETL job updated a table’s schema. To keep your code accurate and type-safe, refresh your external models so queries, APIs, and materialized views reference the correct columns and types.
+  ```bash filename="Terminal" copy
+  moose db pull --connection-string <YOUR_CLICKHOUSE_URL>
+  ```
+  This updates only `EXTERNALLY_MANAGED` models and leaves managed code untouched.
+
+### Automatically run on dev startup (keep local fresh)
+  In active development, schemas can drift faster than you commit updates. Running `db pull` on dev startup helps ensure your local code matches the live schema you depend on.
+  ```bash filename="Terminal" copy
+  export REMOTE_CLICKHOUSE_URL="clickhouse://<user>:<password>@<host>:<port>/<database>"
+  ```
+  Add to `moose.config.toml`:
+  ```toml filename="moose.config.toml" copy
+  [http_server_config]
+  on_first_start_script = "moose db pull --connection-string $REMOTE_CLICKHOUSE_URL"
+  ```
+  This runs once when the dev server first starts. To run after code reloads, use `on_reload_complete_script`. If you run this frequently, prefer HTTP(S) URLs and cache credentials via env/secrets to avoid friction.
+
+### New project from an existing DB
+  If you’re starting with an existing ClickHouse database, bootstrap code with `init --from-remote`, then use `db pull` over time to keep external models fresh:
+  ```bash filename="Terminal" copy
+  moose init my-project --from-remote $REMOTE_CLICKHOUSE_URL --language <typescript|python>
+  ```
+
+  <Callout type="info" title="See the full guide" ctaLink="/moose/getting-started/from-clickhouse" ctaLabel="Get started">
+  Review the full getting started guide to learn more about how to bootstrap a new Moose OLAP project from an existing ClickHouse DB.
+  </Callout>
+
+## Troubleshooting
+
+- **No changes written**: Ensure tables are actually marked as `EXTERNALLY_MANAGED` and names match remote.
+- **Unsupported types**: The CLI will list tables with unsupported types; they’re skipped in the generated file.
```

Comment:
add a CTA to open a github issue or ping on slack to request another type to be supported?